galite-core/src/main/kotlin/org/kopi/galite/visual/Module.kt

```diff
@@ -37,24 +70,123 @@ class Module(val id: Int,
+      description = "!!! $shortname !!!"
+      help = description
+    }
   }
 
-  override operator fun compareTo(module: Module?): Int {
-    TODO()
+  override fun compareTo(module: Module): Int {
+    return if (priority == module.priority) {
+      description.compareTo(module.description)
+    } else {
+      priority - module.priority
+    }
   }
 
-  var description: String? = null
+  /**
+   * return the mnemonic
+   */
+  var icon: Image? = null
+    private set
+
+  /**
+   * return the name of the object which are linked with the module
+   * this object is the name of the class to be executed when this module
+   * is called.
+   */
+  var objectName : String
     private set
-  var help: String? = null
+
+  /**
+   * return description of the module
+   */
+  lateinit var description: String
+    private set
+
+  /**
+   * return the help
+   */
+  lateinit var help: String
     private set
 
+  /**
+   * return the mnemonic
+   */
+  private var smallIcon: Image? = null
+
   companion object {
-    fun getExecutable(data :String?):Executable = TODO()
+    fun getExecutable(objectName: String): Executable {
+      return try {
+        Class.forName(objectName).newInstance() as Executable
+      } catch (iae: IllegalAccessException) {
+        throw VRuntimeException(iae)
+      } catch (ie: InstantiationException) {
+        throw VRuntimeException(ie)
+      } catch (cnfe: ClassNotFoundException) {
+        throw VRuntimeException(cnfe)
+      }
+    }
+
+    fun startForm(ctxt: DBContext,
+                  objectName: String,
+                  description: String,
+                  icon: Image? = null): Executable? {
+      return try {
+        if (ApplicationContext.getDefaults().isDebugModeEnabled()) {
+          System.gc()
+          Thread.yield()
+        }
+        val form: Executable = getExecutable(objectName)
 
+        if (form is VWindow) {
+          form.smallIcon = icon
+        }
+        form.dBContext = ctxt
+        form.doNotModal()
+        form
+      } catch (v: VException) {
+        v.printStackTrace()
+        throw v
+      } catch (t: Throwable) {
+        ApplicationContext.reportTrouble("Form loading",
+                                         "Module.startForm(DBContext ctxt, " +
+                                                 "String object, String description, ImageIcon icon)",
+                                         t.message,
+                                         t)
+        ApplicationContext.displayError(ApplicationContext.getMenu().getDisplay(),
+                MessageCode.getMessage("VIS-00041"))
+        null
+      }
+    }
+
+    // ---------------------------------------------------------------------
+    // DATA MEMBERS
+    // ---------------------------------------------------------------------
     const val ACS_PARENT = 0
     const val ACS_TRUE = 1
     const val ACS_FALSE = 2
   }
-}
 
+  init {
+    this.objectName = objectName!!
```

Comment:
this will lead to NPE
object name could be nullable so dont add (!!)
also you can join initialization with property declaration